src/main/java/github/scarsz/discordsrv/api/commands/SlashCommandProvider.java

```diff
@@ -0,0 +1,24 @@
+package github.scarsz.discordsrv.api.commands;
+
+import github.scarsz.discordsrv.api.ApiManager;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.CommandInteraction;
+import org.bukkit.plugin.Plugin;
+
+import java.util.Set;
+
+/**
+ * Provides {@link PluginCommandData} datum to DiscordSRV's {@link ApiManager}.
+ * {@link Plugin}s implementing this interface are automatically registered.
+ * To manually register a provider, use {@link ApiManager#addSlashCommandProvider(SlashCommandProvider)}.
+ * To handle commands, create a method in a {@link SlashCommandProvider} with the {@link SlashCommand} annotation.
```

Comment:
Had to lookup what datum means, and doesn't seem too fitting there.

Should note the required parameter for the method when using the annotation